server/middleware/security/authentication.js

```diff
@@ -84,9 +84,24 @@ export const checkJwtExpiry = (req, res, next) => {
 export const _authenticateUserByJwt = (req, res, next) => {
   if (!req.jwtPayload) return next();
   const userid = Number(req.jwtPayload.sub);
+  const lastLogin = req.jwtPayload.last;
   User.findByPk(userid)
     .then(user => {
       if (!user) throw errors.Unauthorized(`User id ${userid} not found`);
+      if (req.jwtPayload.last) {
+        if (req.jwtPayload.last === 'firstLogin' && !user.lastLogin) {
+          const now = new Date();
+          user.update({ lastLogin: now });
+        } else {
+          const jwtDate = new Date(lastLogin);
+          const dbLogin = user.lastLogin;
+          if (jwtDate.getTime() !== dbLogin.getTime()) {
+            throw errors.Unauthorized('Invalid login link - multiple uses');
+          }
+          const now = new Date();
+          user.update({ lastLogin: now });
+        }
+      }
```

Comment:
Some code comments would help understand this section. I also think it there is some potential to make it easier to read.

Reply:
For example, one explanation before the `if` and `else` conditions could be interesting.

Reply:
It's now already super clear but I'm wondering if this could be simplified.

I see that in both cases we do the following:

```
const now = new Date();
user.update({ lastLoginAt: now });
```

Maybe we could have that only once by moving it out of the conditions. I can imagine this should then allow some further simplification.

Reply:
I see! Really good point. I'll update that now 😊

Reply:
Updated the code. Let me know when you want to chat through it or if you think it needs some more changes!